action/KafkaProduce/postKafka.js

```diff
@@ -0,0 +1,38 @@
+/**
+ *  Action to post a message to a Message Hub instance
+ *  @param {string} brokers - array of Kafka brokers
+ *  @param {string} topic - topic to produce message
+ *  @param {string} value - content of message
+ *  @param {string} key - message key (optional)
+ *  @param {int}    required_acks - number of acks required before action is complete (optional, default = 1)
+ *  @param {bool}   isJSONData - attempt to parse messages as JSON
+ */
+function main(args) {
+    return validateParameters(args)
+        .then(produceMessage);
```

Comment:
I think you are referring to `produceMessage` not being called with any arguments. If so, notice how it is not being called at all, but instead the function itself is being passed as an argument to `then()`. The result of this is that when the `validateParameters` promise is resolved, `produceMessage` will be called with whatever the `validateParameters` promise resolved to as an argument (in this case, the arguments that were validated by `validateParameters`).

For example:
```javascript
function main() {
    return doTheThing()
        .then(sayTheThing);
}

function doTheThing() {
    return Promise.resolve("Hey!");
}

function sayTheThing(message) {
    console.log("You told me to say: " + message);
}
```

Invoking `main()` eventually prints the message `You told me to say: Hey!`

On the other hand, I might be totally misunderstanding your comment. Let me know...